pac4j-core/src/main/java/org/pac4j/core/client/IndirectClient.java

```diff
@@ -135,9 +137,9 @@ public final C getCredentials(final WebContext context) {
     }
 
     @Override
-    public final RedirectAction getLogoutAction(final WebContext context, final U currentProfile, final String targetUrl) {
+    public final Optional<RedirectAction> getLogoutAction(final WebContext context, final U currentProfile, final String targetUrl) {
         init();
-        return logoutActionBuilder.getLogoutAction(context, currentProfile, targetUrl);
+        return Optional.of(logoutActionBuilder.getLogoutAction(context, currentProfile, targetUrl));
```

Comment:
ofNullable?

Reply:
If we're going to temporarily facade null returns with an Optional until we can replace those null returns with Optional returns (as it appears), then I agree with @victornoel - we should be using ofNullable.

Ideally logoutActionBuilder would return an Optional.

Reply:
yes, if we are going to change things drastically, then also change the return type of everything you think is potentially returing null (and then don't use `ofNullable` in that case of course).